### PR TITLE
[IMP] mail: auto join call

### PR DESCRIPTION
**Pull request by zel-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/public/discuss_public.js [hunk 1]
```diff
@@ -30,6 +31,9 @@ export class DiscussPublic extends Component {
                         null,
                         `/discuss/channel/${this.thread.id}${window.location.search}`
                     );
+                    if (this.thread.defaultDisplayMode === "video_full_screen") {
+                        this.rtc.toggleCall(this.thread, true);
```
Comment (zel-odoo) [inline addons/mail/static/src/new/public/discuss_public.js]: we should have a way to enterFullScreen mode like before. But the problem is from rtc_services, I don't find a good way to put or call the `Call/enterFullScreen()`, previously, we have the model/component so we can call the model directly. But now, we don't have an access to the Call directly, so maybe we should have a prop like `enterWithFullScreen`? I don't know if @ThanhDodeurOdoo likes this or not.

File: addons/mail/static/src/new/public/discuss_public.js [hunk 2]
```diff
@@ -30,6 +31,9 @@ export class DiscussPublic extends Component {
                         null,
                         `/discuss/channel/${this.thread.id}${window.location.search}`
                     );
+                    if (this.thread.defaultDisplayMode === "video_full_screen") {
+                        this.rtc.toggleCall(this.thread, { video: true });
```
Comment (alexkuhn) [inline addons/mail/static/src/new/public/discuss_public.js:35]: I took the opportunity to pass named boolean

File: addons/mail/static/src/new/public/discuss_public.js (hunk 1 above)
Comment (ThanhDodeurOdoo) [reply addons/mail/static/src/new/public/discuss_public.js]: @zel-odoo wouldn't `this.props.thread.defaultDisplayMode` from the call component work?

Comment (zel-odoo) [reply addons/mail/static/src/new/public/discuss_public.js]: I don't think so, this is what we had before and I don't find a good alternative to make the call go into fullScreen
```js
await this.channel.toggleCall({ startWithVideo: true });
await this.threadView.callView.activateFullScreen();
```

Comment (ThanhDodeurOdoo) [reply addons/mail/static/src/new/public/discuss_public.js]: @zel-odoo maybe an event on the bus is the most straightforward

**Pull request merged**